### PR TITLE
Add helper to create log files

### PR DESCRIPTION
**Pull request by dgageot**
Not used yet.

Signed-off-by: David Gageot <david@gageot.net>

### CODE REVIEW

File: pkg/skaffold/logfile/logfile.go [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"regexp"
+)
+
+// Create creates or truncates a file to be used to output logs.
+func Create(name string) (*os.File, error) {
+	root := filepath.Join(os.TempDir(), "skaffold")
```
Comment (briandealwis) [inline pkg/skaffold/logfile/logfile.go:28]: If I perform a sequence of builds, the logs will all be placed in this same directory.  Shouldn't we try to isolate the logs on a per-invocation basis?

Comment (dgageot) [reply pkg/skaffold/logfile/logfile.go:28]: For now, I let the caller give a different name each time

File: pkg/skaffold/logfile/logfile.go [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+	if err := os.MkdirAll(root, 0700); err != nil {
```
Comment (nkubala) [inline pkg/skaffold/logfile/logfile.go:29]: nit: probably don't need the execute bit set on these.

```suggestion
	if err := os.MkdirAll(root, 0600); err != nil {
```

File: pkg/skaffold/logfile/logfile.go [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+		return nil, fmt.Errorf("unable to temp directory %q: %w", root, err)
```
Comment (nkubala) [inline pkg/skaffold/logfile/logfile.go:30]:
```suggestion
		return nil, fmt.Errorf("unable to create temp directory %q: %w", root, err)
```

File: pkg/skaffold/logfile/logfile.go [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+	}
+
+	path := filepath.Join(root, escape(name))
+	return os.OpenFile(path, os.O_RDWR|os.O_CREATE|os.O_TRUNC, 0600)
```
Comment (nkubala) [inline pkg/skaffold/logfile/logfile.go:34]: wouldn't this also imply that successive runs of `skaffold build` would append logs to the same file if the image names matched up? i.e. if caching was disabled

Comment (dgageot) [reply pkg/skaffold/logfile/logfile.go:34]: This code will truncate the log file on each `Create()`. The caller can avoid calling `Create()` with the same name for now.

**Pull request merged**